### ISSUE TITLE
Introduce UUIDs for CelestialAnchors

### ISSUE THREAD

**Issue by JonasGilg** [cosmoscout/cosmoscout-vr#145]
## The Problem
Comparing anchors is mostly done over the centerName, which has problems:
- String comparisons are expensive
- Two anchors with the same center can't be compared this way. For example a SimpleBody and its atmosphere.

## The Solution
Introduce UUIDs for anchors. Maybe use [Boost Uuid](https://www.boost.org/doc/libs/1_64_0/libs/uuid/).

## The Alternatives
Compare anchors by memory address, although that can lead to other problems.

**Comment by Schneegans** [comment]
Could you explain in which places this would be useful? I think the semantics of comparing anchors varies a lot throughout our code base. Sometimes we want to know whether two anchors share the same center, sometimes we want to know whether they share the same frame, sometimes both. Sometimes we even may want to compare their additional transformation.

Are there many places where we really check for physical identity (and where we cannot simply compare a shared_ptr)?

**Comment by TheJhonny007** [comment]
This idea stems from a problem I faced a few months ago. It involves the eclipses. I wanted to avoid self shadowing for eclipse shadows.

There were several problems with that:
- Everything lied within `cs-graphics` so I didn't have access to `cs-scene` and the anchors directly.
- I passed in the centerNames
- Saturns Rings have the same `centerName` as Saturn, so the Rings wouldn't receive a shadow from the Planet

A unique identifier would solve this. I can pass it to modules that shouldn't know about anchors.

> and where we cannot simply compare a shared_ptr

Comparing `shared_ptr`s can be unreliable, since we allow subclassing in plugins and we can't make guarantees about the comparison method.

But you are right it wouldn't replace all comparisons, just the physical equality checks.
